backend/fms_core/migrations/0026_v3_5_0.py

```diff
@@ -0,0 +1,298 @@
+                    WITH RECURSIVE derived AS (
+                        SELECT samplelineage.id AS id, samplelineage.parent_id, samplelineage.child_id, samplelineage.parent_id AS root_sample_id,
+                               samplelineage.process_measurement_id
+                        FROM fms_core_samplelineage samplelineage
+                        WHERE samplelineage.parent_id IN
+                              (SELECT id FROM fms_core_sample WHERE id NOT IN (SELECT child_id FROM fms_core_samplelineage))
+                        UNION ALL
+                             SELECT sl2.id AS id, sl2.parent_id, sl2.child_id, derived.root_sample_id, sl2.process_measurement_id
+                             FROM fms_core_samplelineage sl2
+                             JOIN derived
+                                 ON derived.child_id = sl2.parent_id
+                    )
+                    -- Select samples that are children in some samplelineage
+                    -- Ensure that they are not from samplelineage created from a Transfer process
+                    SELECT derived.child_id AS sample_id, biosample.id AS biosample_id
+                    FROM derived
+                    JOIN fms_core_biosample biosample
+                    ON biosample.root_sample_id = derived.root_sample_id
+                    JOIN fms_core_processmeasurement processmeasurement
+                    ON processmeasurement.id = derived.process_measurement_id
+                    JOIN fms_core_process process
+                    ON process.id = processmeasurement.process_id
+                    JOIN fms_core_protocol protocol
+                    ON protocol.id = process.protocol_id
+                    WHERE protocol.name != 'Transfer'
+                ) t
+                -- Join sample to the sample from the samplelineage in order to get the sample attr to insert into the DerivedSample
+                JOIN fms_core_sample s
+                ON t.sample_id = s.id;
+            """,
+            migrations.RunSQL.noop
+        ),
+        migrations.RunSQL(
+            """
+                -- Create DerivedBySample based on parent-only samples based on SampleLineage
+                INSERT INTO fms_core_derivedbysample (sample_id, derived_sample_id, volume_ratio, created_at, created_by_id, updated_at, updated_by_id, deleted)
+                SELECT DISTINCT(samplelineage.parent_id), derivedsample.id, 1, sample.created_at, sample.created_by_id,
+                       current_timestamp, 1, FALSE
+                FROM fms_core_samplelineage samplelineage
+                JOIN fms_core_sample sample ON sample.id = samplelineage.parent_id
+                JOIN fms_core_derivedsample derivedsample ON derivedsample.sample_id = parent_id
+                JOIN fms_core_processmeasurement processmeasurement ON processmeasurement.id = samplelineage.process_measurement_id
+                WHERE parent_id NOT IN (SELECT child_id FROM fms_core_samplelineage);
+            """
```

Comment:
Might be a good idea to add migrations.RunSQL.noop to be able to rollback to a previous migration if needed.